### ISSUE TITLE
Is there a reason that "Versions" in the Project struct should remain of type []interface{}?

### ISSUE THREAD

**Issue by attack7** [andygrunwald/go-jira#29]
I'm wondering if there's a reason that Version is an array of interface{}, and could it be something more explicit like []Version and it's just not implemented yet?  If so, I'm happy to define it, but I think it would be a breaking change for people who are using it as is.

**Comment by andygrunwald** [comment]
The `Version` field was introduced in #11 (see https://github.com/andygrunwald/go-jira/pull/11/files#diff-4700095d37965ce7b3c2b43fec5d352bR113).

According to the JSON-Mocks for the unit tests (see https://github.com/andygrunwald/go-jira/pull/11/files#diff-a3df89d5900f98d69546f4c705a28007R305) this can be formed into a well defined struct.

The JIRA-Docs (see https://docs.atlassian.com/jira/REST/cloud/#api/2/project-getProject) prove this as well.
This is the schema (available in the docs as well):

```json
"versions": {
            "type": "array",
            "items": {
                "title": "Version",
                "type": "object",
                "properties": {
                    "expand": {
                        "type": "string"
                    },
                    "self": {
                        "type": "string",
                        "format": "uri"
                    },
                    "id": {
                        "type": "string"
                    },
                    "description": {
                        "type": "string"
                    },
                    "name": {
                        "type": "string"
                    },
                    "archived": {
                        "type": "boolean"
                    },
                    "released": {
                        "type": "boolean"
                    },
                    "overdue": {
                        "type": "boolean"
                    },
                    "userStartDate": {
                        "type": "string"
                    },
                    "userReleaseDate": {
                        "type": "string"
                    },
                    "project": {
                        "type": "string"
                    },
                    "projectId": {
                        "type": "integer"
                    },
                    "moveUnfixedIssuesTo": {
                        "type": "string",
                        "format": "uri"
                    },
                    "operations": {
                        "type": "array",
                        "items": {
                            "title": "Simple Link",
                            "type": "object",
                            "properties": {
                                "id": {
                                    "type": "string"
                                },
                                "styleClass": {
                                    "type": "string"
                                },
                                "iconClass": {
                                    "type": "string"
                                },
                                "label": {
                                    "type": "string"
                                },
                                "title": {
                                    "type": "string"
                                },
                                "href": {
                                    "type": "string"
                                },
                                "weight": {
                                    "type": "integer"
                                }
                            },
                            "additionalProperties": false
                        }
                    },
                    "remotelinks": {
                        "type": "array",
                        "items": {
                            "title": "Remote Entity Link",
                            "type": "object",
                            "properties": {
                                "self": {
                                    "type": "string",
                                    "format": "uri"
                                },
                                "name": {
                                    "type": "string"
                                },
                                "link": {}
                            },
                            "additionalProperties": false
                        }
                    }
                },
                "additionalProperties": false
            }
        },
```

Would you be so nice to apply a well defined struct @attack7 ?

**Comment by andygrunwald** [comment]
Solved in #30  :)
Thank you!
